### ISSUE TITLE
show_prediction() doesn't work inside a loop

### ISSUE THREAD

**Issue by calz1** [TeamHG-Memex/eli5#205]
I am running all this in a Jupyter notebook and probably doing something dumb. If I wanted to output a series of textexplainer predictions, what is the right way? 

I have a data setup very similar to newsgroup example in the tutorial [here](http://eli5.readthedocs.io/en/latest/tutorials/black-box-text-classifiers.html#textexplainer) and almost reused the variable names to eliminate a typo as the problem. I am trying to print out the highlighted text explanations for several documents in a row.

When I do this:

```
for i in range(0,2):
    doc = twenty_test_data[i]
    print_prediction(doc)
    print(doc)
    te = TextExplainer(random_state=42)
    te.fit(doc, pipe.predict_proba)
    te.show_prediction(target_names=twenty_train_target_names)
    print('----')
```

I get nothing but two copies of the prediction probabilities, the text of my document, and ---, something like:

> 0.072 P
> 0.928 P
> blah blah blah foo bar baz
> ----
> 0.032 P
> 0.968 P
> foo bar baz blah blah blah 
> ----

When I do **this** which is exactly the same except the TextExplainer _after_ the loop:

```
for i in range(0,2):
    doc = twenty_test_data[i]
    print_prediction(doc)
    print(doc)
    print('----')

te = TextExplainer(random_state=42)
te.fit(doc, pipe.predict_proba)
te.show_prediction(target_names=twenty_train_target_names)
```

then it prints the same as above but with the HTML contribution table and color-coded text at the end because it was the last doc set.

**Comment by kmike** [comment]
Hey,


`te.show_prediction` name may be a bit confusing: it doesn't display the result immediately; for more versatility it returns a display.HTML object. If you have something at the end of a cell, this something is displayed in a notebook - e.g. if you enter "1+2" to the cell, output will be 3. That's the reason a table is shown in your last example - it is just the way display.HTML object represents itself for IPython notebook. 

To show a table inside a loop (or at any other point of time) use `from IPython.display import display` function, as described at  http://eli5.readthedocs.io/en/latest/autodocs/eli5.html#eli5.show_prediction (see "Returns" section):

```py
from IPython.display import display

for i in range(0,2):
    doc = twenty_test_data[i]
    print_prediction(doc)
    print(doc)
    te = TextExplainer(random_state=42)
    te.fit(doc, pipe.predict_proba)
    display(te.show_prediction(target_names=twenty_train_target_names))
    print('----')

```

**Comment by calz1** [comment]
@kmike Thanks, that did it and was admittedly in the docs :( That worked great in outputting multiple items at a time. 

I saw this commit #206 go in a few hours which is really cool because my next attempt was to send that output to a machine-readable JSON for use elsewhere.
